Atomic/syscontainers.py

```diff
@@ -675,7 +678,45 @@ def _amend_values(self, values, manifest, name, image, image_id, destination, pr
         return values
 
     @staticmethod
-    def _are_on_the_same_filesystem(a, b):
+    def _is_repo_on_the_same_filesystem(repo, destdir):
+        """
+        :param repo: Existing ostree repository.
+        :type repo: str
+
+        :param destdir: path to check.  It is created if it doesn't exist.  It must be
+        writeable as we create a temporary file there.
+        :type destdir: str
+        """
+        repo_stat = os.stat(repo)
+        destdir_stat = os.stat(destdir)
+        # Simple case: st_dev is different
+        if repo_stat.st_dev != destdir_stat.st_dev:
+            return False
+
+        try:
+            os.makedirs(destdir)
```

Comment:
Won't this leave a created directory behind if everything succeeds? Not the worst thing in the world if it does, but since the ``dest_file`` is cleaned up later I'd expect this to be cleaned up. If it's meant to stick around maybe change the name of the function as it sounds like it shouldn't have a side effect.

Reply:
I didn't add the rmdir as in any case more containers will still use the same directory (which is hidden in the .storage path).   It is more of a tmp area where we can write temporary files.  In any case, `atomic images prune` takes care of all the temporary files left in the storage (in case the unlink for the dest_file fails)

Reply:
Ok, fair enough.